components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/Utils.kt

```diff
@@ -64,12 +64,20 @@ suspend fun <T> handleFxaExceptions(
 }
 
 /**
- * Helper method that handles [FxaException] and returns a [Boolean] success flag as a result.
+ * Helper method that pre-processes any exceptions for the given
+ *
+ * @param logger object used for logging.
+ * @param operation short description of the operation to be completed. Used in logging.
+ * @param block operation to be completed.
+ *
+ * @return [FxaException] with more details about why the operation failed or
+ *
+ * [null] if the operation completed successfully.
  */
-suspend fun handleFxaExceptions(logger: Logger, operation: String, block: () -> Unit): Boolean {
-    return handleFxaExceptions(logger, operation, { false }, {
+suspend fun handleFxaExceptions(logger: Logger, operation: String, block: () -> Unit): FxaException? {
```

Comment:
You don't need to change this method. Instead, in `sendCommandToDevice` change how `handleFxaExceptions` is used:

```
        val result = handleFxaExceptions(logger, "sending device command", { error -> error }) {
            when (outgoingCommand) {
                is DeviceCommandOutgoing.SendTab -> {
                    account.sendSingleTab(targetDeviceId, outgoingCommand.title, outgoingCommand.url)
                    SyncTelemetry.processFxaTelemetry(account.gatherTelemetry(), crashReporter)
                }
                else -> logger.debug("Skipped sending unsupported command type: $outgoingCommand")
            }
            null
        }
```

And that should be pretty much it!

Reply:
Thanks!